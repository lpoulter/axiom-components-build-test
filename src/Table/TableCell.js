import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';

export default class TableCell extends Component {
  static propTypes = {
    /** Tabular data for this cell */
    children: PropTypes.node,
    className: PropTypes.string,
    /** Sets the horizontal padding around the cell's content */
    horizontalPadding: PropTypes.oneOf(['none', 'medium']),
    /** Marks cell as selected */
    isSelected: PropTypes.bool,
    /** Set text-align */
    textAlign: PropTypes.oneOf(['left', 'right', 'centre']),
    /** Sets the vertical padding around the cell's content */
    verticalPadding: PropTypes.oneOf(['none', 'small', 'medium', 'large']),
  };

  static defaultProps = {
    horizontalPadding: 'medium',
    isSelected: false,
    textAlign: 'left',
    verticalPadding: 'small',
  }

  render() {
    const {
      className,
      children,
      horizontalPadding,
      isSelected,
      textAlign,
      verticalPadding,
      ...rest
    } = this.props;

    const classes = classnames(
      'ax-table__cell',
      `ax-table__cell--align-${textAlign}`,
      `ax-table__cell--horizontal-padding-${horizontalPadding}`,
      `ax-table__cell--vertical-padding-${verticalPadding}`,
      { 'ax-table__cell--selected': isSelected },
      className
    );

    return (
      <Base { ...rest } Component="td" className={ classes }>
        { children }
      </Base>
    );
  }
}
