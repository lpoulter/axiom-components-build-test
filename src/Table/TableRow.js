import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';

export default class TableRow extends Component {
  static propTypes = {
    /** Specifies whether or not the row should have borders */
    borderless: PropTypes.bool,
    /** TableCells */
    children: PropTypes.node,
    className: PropTypes.string,
    /** Apply hover styling */
    hover: PropTypes.bool,
    /** When pCardrovided, applies styling to indicate the TableRow is clickable */
    onClick: PropTypes.func,
  };

  render() {
    const {
      children,
      className,
      borderless,
      hover,
      onClick,
      ...rest
    } = this.props;

    const classes = classnames('ax-table__row', {
      'ax-table__row--hover': hover,
      'ax-table__row--clickable': onClick,
      'ax-table__row--borderless': borderless,
    }, className);

    return (
      <Base className={ classes } { ...rest } Component="tr" onClick={ onClick }>
        { children }
      </Base>
    );
  }
}
