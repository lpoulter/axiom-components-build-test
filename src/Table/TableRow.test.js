import React from 'react';
import renderer from 'react-test-renderer';
import TableCell from './TableCell';
import TableRow from './TableRow';

const getComponent = (props = {}) =>
  renderer.create(
    <TableRow { ...props }>
      <TableCell>
        123456
      </TableCell>
    </TableRow>
  );

describe('TableRow', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without borders', () => {
    const component = getComponent({ borderless: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with clickable styling', () => {
    const component = getComponent({ onClick: () => {} });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
