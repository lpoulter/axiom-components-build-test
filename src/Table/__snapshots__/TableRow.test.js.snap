// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow renders with clickable styling 1`] = `
<tr
  className="ax-table__row ax-table__row--clickable"
  onClick={[Function]}
>
  <td
    className="ax-table__cell ax-table__cell--align-left ax-table__cell--horizontal-padding-medium ax-table__cell--vertical-padding-small"
  >
    123456
  </td>
</tr>
`;

exports[`TableRow renders with defaultProps 1`] = `
<tr
  className="ax-table__row"
>
  <td
    className="ax-table__cell ax-table__cell--align-left ax-table__cell--horizontal-padding-medium ax-table__cell--vertical-padding-small"
  >
    123456
  </td>
</tr>
`;

exports[`TableRow renders without borders 1`] = `
<tr
  className="ax-table__row ax-table__row--borderless"
>
  <td
    className="ax-table__cell ax-table__cell--align-left ax-table__cell--horizontal-padding-medium ax-table__cell--vertical-padding-small"
  >
    123456
  </td>
</tr>
`;
